code/game/objects/effects/temporary_visuals/miscellaneous.dm

```diff
@@ -96,17 +96,29 @@
 /obj/effect/temp_visual/order/Initialize(mapload, faction)
 	. = ..()
 	prepare_huds()
+	var/marker_flags
 	var/hud_type
 	if(faction == FACTION_TERRAGOV)
 		hud_type = DATA_HUD_SQUAD_TERRAGOV
 	else if(faction == FACTION_TERRAGOV_REBEL)
 		hud_type = DATA_HUD_SQUAD_REBEL
 	else if(faction == FACTION_SOM)
 		hud_type = DATA_HUD_SQUAD_SOM
+	if(hud_type == DATA_HUD_SQUAD_TERRAGOV)
+		marker_flags = MINIMAP_FLAG_MARINE
+	else if(hud_type == DATA_HUD_SQUAD_REBEL)
+		marker_flags = MINIMAP_FLAG_MARINE_REBEL
+	else if(hud_type == DATA_HUD_SQUAD_SOM)
+		marker_flags = MINIMAP_FLAG_MARINE_SOM
 	var/datum/atom_hud/squad/squad_hud = GLOB.huds[hud_type]
 	squad_hud.add_to_hud(src)
+	SSminimaps.add_marker(src, src.z, marker_flags, icon_state_on)
 	set_visuals(faction)
 
+/obj/effect/temp_visual/order/Destroy()
+	SSminimaps.remove_marker(src)
```

Comment:
Correct me if im wrong but doesnt SSminimaps self remove on deletion anyway

Reply:
tested and yea